src/peeking_take_while.rs

```diff
@@ -242,10 +242,28 @@ peeking_next_by_clone! { ['a] ::std::str::Bytes<'a> }
 peeking_next_by_clone! { ['a, T] ::std::option::Iter<'a, T> }
 peeking_next_by_clone! { ['a, T] ::std::result::Iter<'a, T> }
 peeking_next_by_clone! { [T] ::std::iter::Empty<T> }
+
 #[cfg(feature = "use_alloc")]
-peeking_next_by_clone! { ['a, T] alloc::collections::linked_list::Iter<'a, T> }
-#[cfg(feature = "use_alloc")]
-peeking_next_by_clone! { ['a, T] alloc::collections::vec_deque::Iter<'a, T> }
+const _: () = {
```

Comment:
Please replace then `const _` with repeated gates.

If we start adopting this pattern, we should do it consistently and not arbitrarily only here.